### PR TITLE
[e2e] Produce message tests, schema registry tests

### PR DESCRIPTION
**Pull request by ChristinaKristane**
- [ ] **Breaking change?** (if so, please describe the impact and migration path for existing applications:)

**What changes did you make?** (Give an overview)

**Is there anything you'd like reviewers to focus on?**

**How Has This Been Tested?** (put an "X" next to an item)

- [ ] No need to
- [ ] Manually(please, describe, when necessary)
- [ ] Unit checks
- [ ] Integration checks
- [x] Covered by existing automation

**Checklist** (put an "X" next to an item, otherwise PR will fail)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [x] My changes generate no new warnings(e.g. Sonar is happy)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged

Check out [Contributing](https://github.com/provectus/kafka-ui/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/provectus/kafka-ui/blob/master/CODE-OF-CONDUCT.md)

### CODE REVIEW

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/ProduceMessagePage.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    private final SelenideElement keyField = $(By.xpath("//div[@id = 'key']/textarea"));
+    private final SelenideElement contentField = $(By.xpath("//div[@id = 'content']/textarea"));
+    private final SelenideElement headersField = $(By.xpath("//div[@id = 'headers']/textarea"));
+    private final SelenideElement sendBtn = $(By.xpath("//button[@type = 'submit']"));
+
+    public ProduceMessagePage setKeyField(String value) {
+        Wait().until(ExpectedConditions.urlContains("message"));
+        keyField.sendKeys(Keys.chord(Keys.DELETE));
+        keyField.setValue(value);
+        return new ProduceMessagePage();
```
Comment (mikementor) [inline kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/ProduceMessagePage.java:22]: `return this;`

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/TopicView.java [hunk 2]
```diff
@@ -99,4 +111,18 @@ public TopicView maxMessageBytesIs(String bytes) {
...
         return this;
     }
-}
\ No newline at end of file
+
+    public boolean isMessageOnPage(String messageName) {
+        Optional<SelenideElement> first = $$(By.xpath("//div[@class ='box']//table//tr//td")).stream()
+                .filter(e -> e.getText().equals(messageName))
+                .findFirst();
+        if(first.isPresent()){
```
Comment (mikementor) [inline kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/TopicView.java:119]: why not `return first.isPresent();`?

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/ProduceMessagePage.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    public ProduceMessagePage setContentFiled(String value) {
```
Comment (mikementor) [inline kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/ProduceMessagePage.java:25]: `setContentField` ?

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/ProduceMessagePage.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        Wait().until(ExpectedConditions.urlContains("message"));
+        contentField.sendKeys(Keys.DELETE);
+        contentField.setValue(value);
+        return new ProduceMessagePage();
+    }
+
+    public ProduceMessagePage setHeaderFiled(String value) {
```
Comment (mikementor) [inline kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/ProduceMessagePage.java:32]: `setHeaderField`?

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/ProduceMessagePage.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        keyField.sendKeys(Keys.chord(Keys.DELETE));
+        keyField.setValue(value);
+        return new ProduceMessagePage();
+    }
+
+    public ProduceMessagePage setContentFiled(String value) {
+        Wait().until(ExpectedConditions.urlContains("message"));
+        contentField.sendKeys(Keys.DELETE);
+        contentField.setValue(value);
+        return new ProduceMessagePage();
```
Comment (mikementor) [inline kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/ProduceMessagePage.java:29]: `return this;`

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/ProduceMessagePage.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    public ProduceMessagePage setHeaderFiled(String value) {
+        headersField.waitUntil(Condition.visible, 10000).setValue(value);
+        return new ProduceMessagePage();
```
Comment (mikementor) [inline kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/ProduceMessagePage.java:34]: `return this;`

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/TopicView.java [hunk 7]
```diff
@@ -1,19 +1,24 @@
...
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.junit.jupiter.api.Assertions;
 import org.openqa.selenium.By;
 
 import static com.codeborne.selenide.Selenide.*;
 
 @ExtensionMethod({WaitUtils.class})
 public class TopicView {
+    private final Logger logger = LogManager.getLogger(this.getClass().getName());
```
[the remark this answers is not in the document]
Comment (mikementor) [reply kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/TopicView.java:22]: still here

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/TopicView.java [hunk 8]
```diff
@@ -47,6 +52,13 @@ public TopicView clickDeleteTopicButton() {
         return this;
     }
 
+    @SneakyThrows
+    public ProduceMessagePage clickOnButton(String buttonName) {
+        logger.info("clickOnButton == '{}'", buttonName);
+        $(By.xpath("//div[@class ='buttons']//*[text()='%s']".formatted(buttonName))).click();
+        return new ProduceMessagePage();
```
Comment (mikementor) [inline kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/TopicView.java:59]: `return this`

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/TopicsList.java [hunk 9]
```diff
@@ -40,6 +40,7 @@ public TopicsList openTopic(String topicName) {
     @SneakyThrows
     public TopicsList isNotVisible(String topicName) {
         By.xpath("//div[contains(@class,'section')]//table").refreshUntil(Condition.visible);
+        Thread.sleep(10000);
```
[the remark this answers is not in the document]
Comment (mikementor) [reply kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/TopicsList.java:43]: ?

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/topics/TopicTests.java [hunk 10]
```diff
@@ -85,4 +106,20 @@ void deleteTopic() {
...
     }
+
+    @SneakyThrows
+    @DisplayName("produce message")
+    @Test
+    void produceMessage(){
+        pages.openTopicsList(SECOND_LOCAL)
+                .isOnPage()
+                .openTopic(TOPIC_TO_UPDATE);
+        Selenide.refresh();
```
Comment (mikementor) [inline kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/topics/TopicTests.java:117]: is there a reason to refresh the page here?

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/topics/TopicTests.java [hunk 11]
```diff
@@ -57,13 +76,15 @@ void updateTopic() {
         pages.openTopicsList(SECOND_LOCAL)
                 .isOnPage()
                 .openTopic(TOPIC_TO_UPDATE);
+        Selenide.refresh();
```
Comment (mikementor) [inline kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/topics/TopicTests.java:79]: why refresh is here?

**Review by mikementor — CHANGES_REQUESTED** [review]

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/topics/TopicTests.java (hunk 11 above)
Comment (ChristinaKristane) [reply kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/topics/TopicTests.java:79]: waiting until page is loaded

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/topics/TopicTests.java (hunk 10 above)
Comment (ChristinaKristane) [reply kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/topics/TopicTests.java:117]: waiting until page is loaded .

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/ProduceMessagePage.java (hunk 3 above)
Comment (ChristinaKristane) [reply kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/ProduceMessagePage.java:25]: we need to enter value to this field for producing message. Produced message has 3 fields (header, content , key)

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/TopicView.java (hunk 8 above)
Comment (ChristinaKristane) [reply kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/TopicView.java:59]: will do

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/ProduceMessagePage.java (hunk 4 above)
Comment (ChristinaKristane) [reply kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/ProduceMessagePage.java:32]: we need to enter value to this field for producing message. Produced message has 3 fields (header, content , key)

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/ProduceMessagePage.java (hunk 1 above)
Comment (ChristinaKristane) [reply kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/ProduceMessagePage.java:22]: will change

**Review by Haarolean — APPROVED** [review]

**Pull request merged**